lab_3_generate_by_ngrams/main.py

```diff
@@ -270,7 +406,39 @@ def run(self, seq_len: int, prompt: str) -> Optional[str]:
         In case of corrupt input arguments or methods used return None,
         None is returned
         """
+        if not (isinstance(seq_len, int) and isinstance(prompt, str) and prompt
+        ):
+            return None
+
+        n_gram_size = self._model.get_n_gram_size()
+        encoded = self._text_processor.encode(prompt)
+
+        if not (encoded and n_gram_size):
+            return None
+
+        text = prompt
+        for i in range(seq_len):
+            tokens = self._model.generate_next_token(encoded[-n_gram_size + 1:])
```

Comment:
Do we have to extract a piece of `encoded` here or is it handled by `generate_next_token`?